src/docfx/lib/YamlUtility.cs

```diff
@@ -20,45 +20,29 @@ internal static class YamlUtility
         public const string YamlMimePrefix = "YamlMime:";
 
         /// <summary>
-        /// Get YamlMime from TextReader
+        /// Get yaml mime type
         /// </summary>
-        public static string ReadMime(TextReader reader)
+        public static string ReadMime(string yaml)
         {
-            var content = ReadHeader(reader);
-            if (!content.StartsWith(YamlMimePrefix, StringComparison.OrdinalIgnoreCase))
+            var header = ReadHeader(yaml);
+            if (header == null || !header.StartsWith(YamlMimePrefix, StringComparison.OrdinalIgnoreCase))
             {
                 return null;
             }
-            return content;
+            return header.Substring(YamlMimePrefix.Length).Trim();
         }
 
         /// <summary>
         /// Get the content of the first comment line
         /// </summary>
-        public static string ReadHeader(TextReader reader)
```

Comment:
Shall we remove Deserialize method with TextReader input as well(JsonUtility and YamlUtility)? Seems not in use except tests

Reply:
I'll do it in another PR.